src/gluonnlp/vocab/bert.py

```diff
@@ -0,0 +1,197 @@
+        A token used to separate sentence pairs for BERT.
+    cls_token : hashable object or None, default '[CLS]'
+        Classification symbol for BERT.
+    reserved_tokens : list of hashable objects or None, default [mask_token, sep_token, cls_token],
+        A list of reserved tokens (excluding `unknown_token`) that will always be indexed, such as
+        special symbols representing padding, beginning of sentence, and end of sentence. It cannot
+        contain `unknown_token` or duplicate reserved tokens. Keys of `counter`, `unknown_token`,
+        and values of `reserved_tokens` must be of the same hashable type. Examples: str, int, and
+        tuple.
+
+    Attributes
+    ----------
+    UNKNOWN_TOKEN : '[UNK]'
+        Static attribute of BERTVocab, unknown token for BERT.
+    PADDING_TOKEN : '[PAD]'
+        Static attribute of BERTVocab, padding token for BERT.
+    MASK_TOKEN : '[MASK]'
+        Static attribute of BERTVocab, mask token for BERT.
+    SEP_TOKEN : '[SEP]'
+        Static attribute of BERTVocab, a token used to separate sentence pairs for BERT.
+    CLS_TOKEN : '[CLS]'
+        Static attribute of BERTVocab, Classification symbol for BERT.
+    embedding : instance of :class:`gluonnlp.embedding.TokenEmbedding`
+        The embedding of the indexed tokens.
+    idx_to_token : list of strs
+        A list of indexed tokens where the list indices and the token indices are aligned.
+    reserved_tokens : list of strs or None
+        A list of reserved tokens that will always be indexed.
+    token_to_idx : dict mapping str to int
+        A dict mapping each token to its index integer.
+    unknown_token : hashable object or None
+        The representation for any unknown token. In other words, any unknown token will be indexed
+        as the same representation.
+    padding_token : hashable object or None
+        The representation for padding token.
+    bos_token : hashable object or None
+        The representation for beginning-of-sentence token.
+    eos_token : hashable object or None
+        The representation for end-of-sentence token.
+    mask_token : hashable object or None, default '[MASK]'
+        The representation for the special token of mask token for BERT.
+    sep_token : hashable object or None, default '[SEP]'
+        a token used to separate sentence pairs for BERT.
+    cls_token : hashable object or None, default '[CLS]'
+    """
+
+    UNKNOWN_TOKEN = '[UNK]'
+
+    PADDING_TOKEN = '[PAD]'
+
+    MASK_TOKEN = '[MASK]'
+
+    SEP_TOKEN = '[SEP]'
+
+    CLS_TOKEN = '[CLS]'
+
+    def __init__(self, counter=None, max_size=None, min_freq=1, unknown_token=UNKNOWN_TOKEN,
+                 padding_token=PADDING_TOKEN, bos_token=None, eos_token=None,
+                 mask_token=MASK_TOKEN, sep_token=SEP_TOKEN, cls_token=CLS_TOKEN,
+                 reserved_tokens=None):
+        if not reserved_tokens:
+            reserved_tokens = [mask_token, sep_token, cls_token]
```

Comment:
Even if user provides a none empty value for `reserved_tokens`, we should still add mask_token, sep_token, cls_token if they are not None.